### PR TITLE
Update C# code for Dodge The Creeps tutorial

### PR DESCRIPTION
**Pull request by KellyThomas**
Also:
 * re-title "Style Guide" to "C# Style Guide"
 * fix a style error on the signals page.

Godot's C# support has changed since these code samples were initially written.
This PR is updating them to Godot's current behaviour and C# API.

This is suitable for both 3.0.6 and current master.

https://github.com/godotengine/godot-demo-projects/pull/288 is a matching PR for a demo project produced with this code.

### CODE REVIEW

**Comment by KellyThomas** [comment]
@exts  
You wrote the initial samples back in April.

Can I please ask you to give this a look over to see if these changes look good.

**Comment by KellyThomas** [comment]
I've rebased this to resolve some conflicts

File: getting_started/step_by_step/your_first_game.rst [hunk 1]
```diff
@@ -1198,5 +1201,6 @@ all the way to ``0``.
 Project files
 -------------
 
-You can find a completed version of this project here:
-https://github.com/kidscancode/Godot3_dodge/releases
+You can find a completed version of this project at these locations:
+ - https://github.com/kidscancode/Godot3_dodge/releases
+ - https://github.com/godotengine/godot-demo-projects
```
[the remark this answers is not in the document]
Comment (cbscribe) [reply getting_started/step_by_step/your_first_game.rst]: My only problem with linking to demo-projects is that it's a huge repo containing many projects. The reader has to clone the whole repo for the one project, which can be overwhelming.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**